src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -1098,6 +1098,194 @@ private static Type GetCommonType(Type minType, Type maxType)
     }
 
     /// <summary>
+    /// Base class for postive number valididation attributes
+    /// </summary>
+    /// <remarks>
+    /// Base class used by postive and non negative attribute
+    /// <seealso cref="ValidatePositiveAttribute"/>,
+    /// <seealso cref="ValidateNonNegativeAttribute"/>
+    /// </remarks>
+    [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property)]
+    public class ValidatePositiveBaseAttribute : ValidateEnumeratedArgumentsAttribute
+    {
+        /// <summary>
+        /// Minimum allowed value for the attribute
+        /// </summary>
+        public int minValue;
```

Comment:
Try my suggestion to use `dynamic` - it might just work. Behind the scenes, the C# compiler will generate code calling the correct operator *after* testing the type of `val`.